internal/common/db.go

```diff
@@ -288,3 +290,39 @@ func ToSqlList(strs []string) string {
 	res = fmt.Sprintf("('%s')", res)
 	return res
 }
+
+func CreateTransactionWithDefer(db *gorm.DB, txSuccess *bool) (func(), *gorm.DB, error) {
+	tx := db.Begin()
+	if tx.Error != nil {
+		return nil, nil, tx.Error
+	}
+	tx = transaction.AddForUpdateQueryOption(tx)
+	deferFunc := func() {
+		if !*txSuccess {
+			tx.Rollback()
+		}
+		if r := recover(); r != nil {
+			tx.Rollback()
+		}
+	}
+	return deferFunc, tx, nil
+}
+
+func CreateInfraEnvForCluster(db *gorm.DB, cluster *Cluster) error {
```

Comment:
What about things like static network config, ignition overrides, and ntp sources?

Reply:
static network is passed as a parameter only during GenerateISO request, and we are setting it there into the InfraEnv. 
ntp is not used as part of Discovery generation at all.
ignition overrides in V1 is part of cluster update. we are updating Infraenv with relevant value before calling FormatDiscoveryIgnitionFile. In V2, it will be updating on the InfraEnv directly in a dedicated API

Reply:
not sure i understand your comment:  where am i changing them in this PR?

Reply:
I'm saying that we'll need a way to set static networking that isn't generating the image when generating the image outside of assisted service.

But that doesn't need to be handled in this PR. I'll resolve this comment.